src/Microsoft.ML.Data/Scorers/FeatureContributionCalculation.cs

```diff
@@ -269,13 +233,9 @@ private ValueGetter<ReadOnlyMemory<char>> GetTextValueGetter<TSrc>(Row input, in
                     {
                         featureGetter(ref features);
                         map(in features, ref contributions);
-                        var indices = new Span<int>();
-                        var values = new Span<float>();
-                        if (contributions.IsDense)
-                            Utils.GetIdentityPermutation(contributions.Length).AsSpan().CopyTo(indices);
-                        else
-                            contributions.GetIndices().CopyTo(indices);
-                        contributions.GetValues().CopyTo(values);
+                        var editor = VBufferEditor.CreateFromBuffer(ref contributions);
+                        var indices = contributions.IsDense ? Utils.GetIdentityPermutation(contributions.Length) : editor.Indices;
+                        var values = editor.Values;
                         var count = values.Length;
                         var sb = new StringBuilder();
                         GenericSpanSortHelper<int>.Sort(indices, values, 0, count);
```

Comment:
>indices, values [](start = 56, length = 15)

These need to be reversed: We want to sort by the values, not by the indices.

Reply:
Ok will change that! Thank you 

---
In reply to: [243368410](https://github.com/dotnet/machinelearning/pull/1847#discussion_r243368410) [](ancestors = 243368410)